fabtests/ubertest/cq.c

```diff
@@ -359,12 +362,28 @@ int ft_comp_rx(int timeout)
 {
 	int ret = 0;
 	size_t cur_credits = ft_rx_ctrl.credits;
+	int cq_ret;
 
 	if (ft_use_comp_cntr(test_info.comp_type)) {
 		ret = ft_cntr_x(rxcntr, &ft_rx_ctrl, timeout);
 		if (ret < 0)
 			return ret;
+
+		if (ft_generates_rx_comp()) {
+			do {
+				cq_ret = ft_comp_x(rxcq, &ft_rx_ctrl, "rxcq", 0);
+				if (cq_ret < 0)
+					return cq_ret;
+			} while (cq_ret > 0);
+		}
+		if (test_info.test_class & (FI_MSG | FI_TAGGED)) {
```

Comment:
Because the "credits" refer to posted sends or posted receives. Any tx completion will clear up that posted send but not only msg or tagged functions will consume a posted buffer
(though there is the weird case of writedata + RX_CQ_DATA mode bit in which case we need to add to the credits which is rather tricky to get around, specifically on the bandwitdh test, and I'm still trying to think of a good solution for this scenario)